clang/lib/DPCT/APINamesMathRewrite.inc

```diff
@@ -566,3 +566,239 @@ MATH_API_REWRITER_DEVICE(
+                                                  "ext::intel::math::hneg2",
+                                              ARG(0)))),
+        EMPTY_FACTORY_ENTRY("__hneg2"), UNARY_OP_FACTORY_ENTRY("__hneg2",
+                                        UnaryOperatorKind::UO_Minus,
+                                        makeCallArgCreatorWithCall(0))))
+
+MATH_API_REWRITER_DEVICE(
+    "__hadd2",
+    MATH_API_DEVICE_NODES(
+        EMPTY_FACTORY_ENTRY("__hadd2"),
+        HEADER_INSERT_FACTORY(
+            HeaderType::HT_SYCL_Math,
+            CALL_FACTORY_ENTRY("__hadd2", CALL(MapNames::getClNamespace() +
+                                                  "ext::intel::math::hadd2",
+                                              ARG(0), ARG(1)))),
+        EMPTY_FACTORY_ENTRY("__hadd2"), BINARY_OP_FACTORY_ENTRY("__hadd2",
+                                        BinaryOperatorKind::BO_Add,
+                                        makeCallArgCreatorWithCall(0),
+                                        makeCallArgCreatorWithCall(1))))
+
+MATH_API_REWRITER_DEVICE(
+    "__h2div",
+    MATH_API_DEVICE_NODES(
+        EMPTY_FACTORY_ENTRY("__h2div"),
+        HEADER_INSERT_FACTORY(
+            HeaderType::HT_SYCL_Math,
+            CALL_FACTORY_ENTRY("__h2div", CALL(MapNames::getClNamespace() +
+                                                  "ext::intel::math::h2div",
+                                              ARG(0), ARG(1)))),
+        EMPTY_FACTORY_ENTRY("__h2div"), BINARY_OP_FACTORY_ENTRY("__h2div",
+                                        BinaryOperatorKind::BO_Div,
+                                        makeCallArgCreatorWithCall(0),
+                                        makeCallArgCreatorWithCall(1))))
+
+MATH_API_REWRITER_DEVICE(
+    "__hdiv",
+    MATH_API_DEVICE_NODES(
+        EMPTY_FACTORY_ENTRY("__hdiv"),
+        HEADER_INSERT_FACTORY(
+            HeaderType::HT_SYCL_Math,
+            CALL_FACTORY_ENTRY("__hdiv", CALL(MapNames::getClNamespace() +
+                                                  "ext::intel::math::hdiv",
+                                              ARG(0), ARG(1)))),
+        EMPTY_FACTORY_ENTRY("__hdiv"), BINARY_OP_FACTORY_ENTRY("__hdiv",
+                                        BinaryOperatorKind::BO_Div,
+                                        makeCallArgCreatorWithCall(0),
+                                        makeCallArgCreatorWithCall(1))))
+
+MATH_API_REWRITER_DEVICE(
+    "h2div",
+    MATH_API_DEVICE_NODES(
+        EMPTY_FACTORY_ENTRY("h2div"),
+        HEADER_INSERT_FACTORY(
+            HeaderType::HT_SYCL_Math,
+            CALL_FACTORY_ENTRY("h2div", CALL(MapNames::getClNamespace() +
+                                                  "ext::intel::math::h2div",
+                                              ARG(0), ARG(1)))),
+        EMPTY_FACTORY_ENTRY("h2div"), BINARY_OP_FACTORY_ENTRY("h2div",
+                                        BinaryOperatorKind::BO_Div,
```

Comment:
I have not found test case for L648 emulation way,  need enable the test in cuda-math-intrinsics.cu line 67.

Reply:
Add the new test case clang/test/dpct/cuda-math-intrinsics1.cu as the hdiv and h2div are 8.0 only.